lib/models/addon.js

```diff
@@ -873,15 +877,32 @@ let addonProto = {
     @return {Tree} The return tree has the same contents as the input tree, but is moved so that the `app/styles/` path is preserved.
   */
   treeForStyles(tree) {
+
+    if (!this.name || this.name === 'superFn') {
+      throw new Error('needs to call _super.treeForStyles.call(...)');
+    }
+
     this._requireBuildPackages();
 
     if (!tree) {
       return tree;
     }
 
-    return new Funnel(tree, {
-      destDir: 'app/styles',
-      annotation: `Addon#treeForStyles (${this.name})`,
+    const name = this.moduleName();
+
+    const addonStyles = new Funnel(tree, {
+      allowEmpty: true,
+      srcDir: this.treePaths.styles,
+      destDir: '.',
+    });
```

Comment:
It seems this is to support addons exposing the styles tree with a `app/styles` (or `src/ui/styles`) folder structure, while previously the Funnel was only taking styles exposed at the root level and moved them to the `destDir` (`only `app/styles` before). So basically you could expose `foo.scss` as `/foo.scss` or as `app/styles/foo.scss`, is that right?

Is so, why is that needed? As this seems to expand the public API contract of the `treeForStyles()` hook (though the public API of those hooks was always kinda fuzzy, AFAICT). Or is it just to support new MU addons?

Also does this rely on addons calling `super` in their `treeForStyles()`? Which some (many?) didn't so far:
* https://github.com/kaliber5/ember-bootstrap/blob/master/index.js#L193-L199
* https://github.com/embermap/ember-cli-tailwind/blob/master/index.js#L69-L77

Reply:
Ah, what I wanted to achieve here was some way to support ember-component-css with addon components out of the box. Therefore I also wanted to merge in all styles, which are not in app/styles...

mmmm, I wasnt aware that many do not call super here... okay, I guess its not so a good idea to add important logic here.
Then i move the logic somewhere else.

Reply:
So, what about moving the logic to `getStyles?
https://github.com/ember-cli/ember-cli/pull/8146/files#diff-392bb207fea8bcbeb4a084bb445f7cc7R1077

Reply:
What i also saw is that root level was already merged into app/styles, so both ways would have worked before?
What i saw is that then there is an app/styles in app/styles folder. But will recheck

Reply:
Okay, so after reviewing.
the styles tree is build here 
https://github.com/ember-cli/ember-cli/blob/master/lib/models/addon.js#L614
which takes the path from
https://github.com/ember-cli/ember-cli/blob/master/lib/models/addon.js#L208

So, all files in <addon>/app/styles will be in the root, when in arrives at `treeForStyles`, but addons could overwrite it and merge in styles, either at root or to `app/styles`, then after calling `this._super.treeForStyles`, root level styles would be moved to the `app/styles`, and `app/styles`, would also still work, but will be duplicated. So to make sure that we dont get into that duplicate state, I dont move app/styles anymore, only root styles.